networks/p2p/discover/node.go

```diff
@@ -208,8 +216,19 @@ func parseComplete(rawurl string) (*Node, error) {
 	if tcpPort, err = strconv.ParseUint(u.Port(), 10, 16); err != nil {
 		return nil, errors.New("invalid port")
 	}
-	udpPort = tcpPort
+	// Extract subport from query
 	qv := u.Query()
+	if qv.Get("subport") != "" {
+		for _, subport := range qv["subport"] {
```

Comment:
Why is this loop needed?

Reply:
Yes, you are right.
Right now, multi-channel server only accept 2 ports. However, it might improve and might use more than 2 ports. I just made `parseNode` to allow more than one port beforehand.